orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -109,11 +108,26 @@ public OrderAggregateDto getOrderWithItems(final String orderId) {
             List<OrderItem> items = this.orderItemService.findByOrderId(orderId);
             List<OrderItemMeta> metas = this.orderItemMetaService.findByOrderId(orderId);
             orderAggregateDto = orderTransformerUtils.transformToOrderAggregateDto(order, items, metas);
+
+            FlowStep flowStep = getFlowStepsByOrderId(orderId);
```

Comment:
Could the repository return optional list? I think it would be possible?

Reply:
And could you make service to flowSteps like in orderItemService etc

Reply:
Good point, I made the service for these. I don't think it makes sense to return a optional list, because the repository will return a list empty list if no results are found. I could change the method **getFlowStepsByOrderId(..)** to return optional flow step.